packages/cysync-core/src/hooks/useDeviceTask.ts

```diff
@@ -78,7 +82,7 @@ export function useDeviceTask<T>(
       setIsRunning(false);
     }
 
-    return error;
+    return { error, result };
```

Comment:
This requires change at `packages\cysync-core\src\hooks\useDeviceUpdate.ts line 71`.